vm/smx-v1-image.cpp

```diff
@@ -756,3 +756,153 @@ SmxV1Image::LookupLine(uint32_t addr, uint32_t* line)
   *line = debug_lines_[low].line + 1;
   return true;
 }
+
+size_t
+SmxV1Image::NumFiles() const
+{
+  return debug_files_.length();
+}
+
+const char*
+SmxV1Image::GetFileName(size_t index) const
+{
+  if (index >= debug_files_.length())
+    return nullptr;
+
+  if (debug_files_[index].name >= debug_names_section_->size)
+    return nullptr;
+
+  return debug_names_ + debug_files_[index].name;
+}
+
+template <typename SymbolType, typename DimType>
+bool
+SmxV1Image::getFunctionAddress(const SymbolType* syms, const char* function, ucell_t* funcaddr, uint32_t& index)
+{
+  const uint8_t* cursor = reinterpret_cast<const uint8_t *>(syms);
+  const uint8_t* cursor_end = cursor + debug_symbols_section_->size;
+  for (uint32_t i = index; i < debug_info_->num_syms; i++) {
+    if (cursor + sizeof(SymbolType) > cursor_end)
+      break;
+
+    const SymbolType *sym = reinterpret_cast<const SymbolType *>(cursor);
+    if (sym->ident == sp::IDENT_FUNCTION &&
+      sym->name < debug_names_section_->size &&
+      !strcmp(debug_names_ + sym->name, function))
+    {
+      *funcaddr = sym->addr;
+      return true;
+    }
+
+    if (sym->dimcount > 0)
+      cursor += sizeof(DimType) * sym->dimcount;
+    cursor += sizeof(SymbolType);
+  }
+  return false;
+}
+
+bool
+SmxV1Image::LookupFunctionAddress(const char* function, const char* file, ucell_t* funcaddr)
+{
+  uint32_t index = 0;
+  const char* tgtfile;

```

Comment:
nit: move tgtfile down to where it's actually used. Same for index. (We're not using C :p)